content/blue-green-deploys-dbt-snowflake.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "Blue Green Deploys Dbt Snowflake"
+date: 2020-05-02T09:24:24-04:00
+draft: true
+---
+
+# Blue Green deploys in Snowflake + DBT
+ #blog #blogpost #draft
+
+As the maintainer of the Analytics stack at SnapTravel, I have gotten a variety  of experience with the Modern Data stack. 
+
+One of the recent challenges the data team faces is: how to move fast with the ever changing business needs and fast growing big data in delivering reliable business insights. I thought using DBT incremental runs for event stream datasets and using appropriate snowflake warehouse sizes was enough to solve the problem. I was wrong, recently, after analyzing past few downtimes in our BI layer, I realized that all of them are related to backfilling  incremental models on billions of rows. Here, I am going to describe the problem and a proposed approach to solve it.
```

Comment:
just a note for self: the downtime is caused by:

1. i push a change and initiate a full refresh
2. i push a change and don't initiate a full refresh. downstream tables/views break (e.g. referencing cols that don't exist), happened a few times and lasted for a weekend